lib/utilities/walk-up-path.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+var path = require('path');
+
+function walkUp(thePath) {
```

Comment:
Hokay, so:
- Let's move away from recursion, can be done in a while loop.
- Doesn't address leading slashes, only relative paths.

Here's a quick sketch of a loop that does the right thing but in which I didn't actually sort out the operations to make it sensible:

```
function walkUp(thePath) {
  var paths = [];
  var lastPath = thePath;
  var nextPath;

  while (true) {
    nextPath = path.dirname(lastPath);
    if (nextPath === lastPath) {
      paths.pop();
      break;
    } 
    paths.push(nextPath);
    lastPath = nextPath;
  }

  return paths;
}
```